src/dataflow/internal/process/functions/source.ts

```diff
@@ -28,6 +28,7 @@ export function processSourceCall<OtherInfo>(functionCall: RFunctionCall<OtherIn
 		const path = removeTokenMapQuotationMarks(sourceFile.lexeme)
 		const request = sourceProvider.createRequest(path)
 
+		// TODO we shouldn't skip a re-analysis *always*, just when it's a cycle - right?
```

Comment:
*Note to self*: It would make a lot more sense for this to, instead of storing a dictionary, just storing the "path" of files that we went down to get to the current source call. If the path already contains this file somewhere in it, it means we're in a loop!